qfdmo/management/commands/import_product_from_qfdmod.py

```diff
@@ -0,0 +1,207 @@
+    product_names = [
+        word.lower().strip()
+        for word in qfdmod_product["Synonymes_existants"].split("/")
+        if word.lower().strip()
+    ]
+    product_names.append(qfdmod_product["Nom"].lower().strip())
+    product_names = list(set(product_names))
+    return product_names
+
+
+def _choose_sous_categorie(sous_categories, product_name):
+    choosen_sous_categorie = None
+    while choosen_sous_categorie is None:
+        input_text = f"\nPour le produit : {product_name}"
+        input_text += "\nChoisissez la categorie adéquat ?\n\n"
+        input_text += "\n".join(
+            [
+                f"{i + 1} : {prop.libelle} ({prop.categorie.libelle})"
+                for i, prop in enumerate(sous_categories)
+            ]
+        )
+        input_text += "\n0 : Ignorer\n\n"
+        input_text += "\n\nVotre choix : "
+        choosen_sous_categorie = input(input_text)
+        if not choosen_sous_categorie.isnumeric() or int(
+            choosen_sous_categorie
+        ) not in range(0, len(sous_categories) + 1):
+            choosen_sous_categorie = None
+    if choosen_sous_categorie == "0":
+        return None
+    return sous_categories[int(choosen_sous_categorie) - 1]
+
+
+def _create_or_update_object_from_product(
+    product_id, product_names, sous_categories, result_in_details
+):
+    """
+    If one category found,
+    Create unknow product using this categories
+    Update identifiant_qfdmod if product already exist
+    If more than one, ask user to choose the right one
+    """
+    for product_name in product_names:
+        obj = Objet.objects.filter(
+            Q(libelle__iexact=product_name) | Q(code__iexact=product_name)
+        ).first()
+        if obj:
+            if obj.identifiant_qfdmod and obj.identifiant_qfdmod != product_id:
+                result_in_details["multi_assignments"].append(product_name)
+            obj.identifiant_qfdmod = product_id
+            obj.save()
+        else:
+            # choose categories
+            if len(sous_categories) > 1:
+                sous_categorie = _choose_sous_categorie(sous_categories, product_name)
+                if sous_categorie is None:
+                    continue
+            else:
+                sous_categorie = sous_categories[0]
+            Objet.objects.create(
+                libelle=product_name,
+                code=product_name,
+                identifiant_qfdmod=product_id,
+                sous_categorie=sous_categorie,
+            )
+    return result_in_details
+
+
+def _should_create_products(
+    qfdmod_product, objects_from_names, product_names, sous_categories
+):
+    product_name = qfdmod_product["Nom"]
+    collect_names_from_objects = [
+        objet.libelle.lower().strip() for objet in objects_from_names
+    ]
+    collect_names_from_objects += [
+        objet.code.lower().strip() for objet in objects_from_names
+    ]
```

Comment:
```suggestion
    collect_names_from_objects = [
        objet.libelle.lower().strip() for objet in objects_from_names
    ] + [
        objet.code.lower().strip() for objet in objects_from_names
    ]
```

je pinaille, mais c'est p-e plus lisible

Reply:
👍